### ISSUE TITLE
Tests fail to load with latest version of a newly generated addon from ember-cli#master

### ISSUE THREAD

**Issue by cowboyd** [switchfly/ember-cli-mocha#93]
Using ember-cli/ember-cli@a2a9a886e1 and trying to create a fresh addon using `ember-cli-mocha`, the tests fail to load.

```
~/C/Ember ❯❯❯ ember addon try-ember-cli-mocha-with-master
version: 2.2.0-beta.1-master-a2a9a886e1
installing addon
  create .bowerrc
  create .editorconfig
  create .ember-cli
  create .jshintrc
  create .travis.yml
  create .watchmanconfig
  create README.md
  create tests/dummy/app/app.js
  create tests/dummy/app/components/.gitkeep
  create tests/dummy/app/controllers/.gitkeep
  create tests/dummy/app/helpers/.gitkeep
  create tests/dummy/app/index.html
  create tests/dummy/app/models/.gitkeep
  create tests/dummy/app/router.js
  create tests/dummy/app/routes/.gitkeep
  create tests/dummy/app/styles/app.css
  create tests/dummy/app/templates/application.hbs
  create tests/dummy/app/templates/components/.gitkeep
  create bower.json
  create tests/dummy/config/environment.js
  create ember-cli-build.js
  create .gitignore
  create package.json
  create tests/dummy/public/crossdomain.xml
  create tests/dummy/public/robots.txt
  create testem.json
  create tests/.jshintrc
  create tests/helpers/destroy-app.js
  create tests/helpers/module-for-acceptance.js
  create tests/helpers/resolver.js
  create tests/helpers/start-app.js
  create tests/index.html
  create tests/integration/.gitkeep
  create tests/test-helper.js
  create tests/unit/.gitkeep
  create vendor/.gitkeep
  create LICENSE.md
  create addon/.gitkeep
  create config/ember-try.js
  create config/environment.js
  create app/.gitkeep
  create index.js
  create .npmignore
Successfully initialized git.
Installed packages for tooling via npm.
Installed browser packages via Bower.
~/C/Ember ❯❯❯ cd try-ember-cli-mocha-with-master
~/C/E/try-ember-cli-mocha-with-master git:master ❯❯❯ ember install ember-cli-mocha
version: 2.2.0-beta.1
Installed packages for tooling via npm.
installing ember-cli-mocha
[?] Overwrite tests/test-helper.js? Yes, overwrite
  overwrite tests/test-helper.js
  install bower packages ember-mocha, ember-cli-test-loader, ember-cli-shims
Installing browser packages via Bower...
  cached git://github.com/ember-cli/ember-cli-test-loader.git#0.2.2
  cached git://github.com/ember-cli/ember-cli-shims.git#0.0.6
  cached git://github.com/switchfly/ember-mocha-builds.git#0.8.8
  cached git://github.com/mochajs/mocha.git#2.2.5
  cached git://github.com/teddyzeenny/ember-mocha-adapter.git#0.3.1
  cached git://github.com/chaijs/chai.git#2.3.0
  conflict Unable to find suitable version for ember-cli-test-loader
    1) ember-cli-test-loader 0.2.1
    2) ember-cli-test-loader 0.2.2
[?] Answer: 2!
  resolution Saved ember-cli-test-loader#0.2.2 as resolution
Installed browser packages via Bower.
  uninstall package ember-cli-qunit
Uninstalled packages for tooling via npm.
Installed addon package.
~/C/E/try-ember-cli-mocha-with-master git:master ❯❯❯ ember t                                                                                                                       
version: 2.2.0-beta.1
Built project successfully. Stored in "/Users/cowboyd/Code/Ember/try-ember-cli-mocha-with-master/tmp/class-tests_dist-JBVbBaTC.tmp".
^Cnot ok 1 Error
    ---
        message: >
            Received SIGINT signal
    ...

1..1
# tests 1
# pass  0
# fail  1
```

The test suite actually hangs until it's manually killed. If you run the test server, then testem prints this to the console:

```
Error: Assertion Failed: The tests file was not loaded. Make sure your tests index.html includes "assets/tests.js". at http://localhost:7357/assets/vendor.js, line 16498
ReferenceError: Can't find variable: describe at http://localhost:7357/assets/test-support.js, line 8561
TypeError: undefined is not an object (evaluating 'Runner.prototype') at http://localhost:7357/testem.js, line 419
```

The developer console shows the following error:

![dummy_tests](https://cloud.githubusercontent.com/assets/4205/11815269/962a2e76-a310-11e5-851e-4771d44c5b2b.png)

**Comment by rwjblue** [comment]
Thanks for reporting. I believe it is related to https://github.com/switchfly/ember-cli-mocha/issues/92, but will dig in tonight.

**Comment by cowboyd** [comment]
Ok, thanks! I added a bit more diagnostic output, but sounds like you're already familiar with it. Let me know if there's anything I can do to help.

**Comment by rwjblue** [comment]
@cowboyd - If you could throw that sample repo up, it will make things a bit quicker when I dig in tonight.

**Comment by cowboyd** [comment]
https://github.com/cowboyd/try-ember-cli-mocha-with-master cheers!

**Comment by mattgrannary** [comment]
Having the same issue, seems like ember-mocha and ember-mocha-adapter are included in the compiled test-support.js file before mocha is initialized.

**Comment by devinus** [comment]
Same issue after upgrading to ember-cli 2.0.0-beta.1.

**Comment by mkempe** [comment]
Fix (works with  2.2.0-beta.2), see #95

**Comment by ballPointPenguin** [comment]
I also experienced this problem earlier.
This issue is resolved for me when using ember-cli#2.3.0-beta.1 as well as master (#bfd6e63)

**Comment by mkempe** [comment]
I think we can close this issue, because it was fixed with https://github.com/ember-cli/ember-cli/pull/5274 aka Ember CLI `2.2.0-beta.3` (https://github.com/ember-cli/ember-cli/releases/tag/v2.2.0-beta.3).

@cowboyd Can you please close this issue? Thanks
